go/network/auth.go

```diff
@@ -0,0 +1,120 @@
+package network
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	pb "go.gazette.dev/core/broker/protocol"
+	"google.golang.org/grpc/metadata"
+)
+
+// verifyAuthorization ensures the request has an authorization which
+// is valid for capability NETWORK_PROXY to `taskName`.
+func verifyAuthorization(req *http.Request, verifier pb.Verifier, taskName string) error {
+	var bearer = req.Header.Get("authorization")
+	if bearer != "" {
+		// Pass.
+	} else if cookie, err := req.Cookie(AuthCookieName); err == nil {
+		bearer = fmt.Sprintf("Bearer %s", cookie.Value)
+	} else {
+		return errors.New("missing authorization")
+	}
+
+	var _, cancel, claims, err = verifier.Verify(
+		metadata.NewIncomingContext(
+			req.Context(),
+			metadata.Pairs("authorization", bearer),
+		),
+		0, // TODO(johnny): Should be pf.Capability_NETWORK_PROXY.
+	)
+	if err != nil {
+		return err
+	}
+	cancel() // We don't use the returned context.
+
+	/* TODO(johnny): Inspect claims once UI is updated to use /authorize/user/task API.
```

Comment:
just to confirm my uderstanding is that we won't actually verify authZ at this stage, but it won't actually allow unauthorized access because this endpoint can't be reached in our current public data plane. So we'll need to update the UI to use the new authorize api, and then enforce authZ here before setting up any of the new private data planes. That sound right?

Reply:
Yep that's right